### PR TITLE
feat: Add hugo built pages

### LINKED ISSUE #14: Hugo Built Pages In NextJs
The current structure requires a hard navigate to `https://bitcointranscripts.com`

For example, selecting a category (Developer Tools), links to `https://btctranscripts.com/tags/#developer-tools`
Screenshot 2024-09-03 at 12 49 40 PMScreenshot 2024-09-03 at 12 51 50 PM

We'd like our incremental development to work without hard navigation to `https://bitcointranscripts.com`, from a user perspective it should be on the same domain.

Proposal:
- Pull [gh-pages](https://github.com/bitcointranscripts/bitcointranscripts.github.io/tree/gh-pages) branch from [bitcointrasnscripts current repo](https://github.com/bitcointranscripts/bitcointranscripts.github.io) as a submodule into nextjs `/public` directory
- setup gh workflow on `bitcointrasnscripts` on push to `ghpages` branch to update submodules on this registry repo, commit and push.

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**
closes #14 
~~Do not merge.~~
~~needs rebase after #7~~

Rebased!

### CODE REVIEW

File: next.config.mjs [hunk 1]
```diff
@@ -1,5 +1,20 @@
 /** @type {import('next').NextConfig} */
 import { withContentlayer } from "next-contentlayer2";
-const nextConfig = {};
+const nextConfig = {
+  rewrites: async () => {
+    return {
+      fallback: [
+        {
+          source: "/plugins/:path*",
+          destination: "/gh-pages/plugins/:path*",
```
Comment (kouloumos) [inline next.config.mjs:9]: what's up with this rule? I don't understand its usage

File: next.config.mjs [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+          source: "/:path*",
+          destination: "/gh-pages/:path*/index.html",
+        },
+      ]
+    }
+  },
+};
 
-export default withContentlayer(nextConfig);
+export default withContentlayer(nextConfig);
```
Comment (kouloumos) [inline next.config.mjs:20]: nit: empty line at the end

**Review by kouloumos — COMMENTED** [review]
Awesome! 

Only one observation, in the "Explore Transcripts" section, the urls with anchors seems to not be working correctly. It redirects just to the `/tags` page instead of the specific section.

Screencastfrom2024-09-0610-57-19-ezgif com-video-to-gif-converter

**Comment by kouloumos** [comment]
The plan is to merge this and then work on 
> setup gh workflow on bitcointrasnscripts on push to ghpages branch to update submodules on this registry repo, commit and push.

?

**Review by IgboPharaoh — APPROVED** [review]
LGTM

**Comment by Emmanuel-Develops** [comment]
> Awesome!
> 
> Only one observation, in the "Explore Transcripts" section, the urls with anchors seems to not be working correctly. It redirects just to the `/tags` page instead of the specific section.
> 

This issue as been documented in #20 
For incremental building, since this issue is isolated to categories, it should not be a blocker for the rest of the hugo pages available.

**Pull request merged**